server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/DoubleValuesSource.java

```diff
@@ -158,7 +158,7 @@ public void collect(int doc, long bucket) throws IOException {
     }
 
     @Override
-    LeafBucketCollector getLeafCollector(Comparable value, LeafReaderContext context, LeafBucketCollector next) {
+    LeafBucketCollector getLeafCollector(Comparable<?> value, LeafReaderContext context, LeafBucketCollector next) {
```

Comment:
In cases like this, shouldn't use `Comparable<Double>` here so we can enforce this at compile time instead of adding an explicit runtime check?

Reply:
Ah, I see, we need to update this method on `SingleDimensionValuesSource` to use the proper generic type argument instead of a wildcard. We probably have missed this in other areas as well. We should go back and check where we've used wildcards that there isn't already a proper type argument we _should_ be using instead.